docs/fml-front-end-format.md

```diff
@@ -0,0 +1,198 @@
+- Support for multiple documents within one file, so each feature could be a separate "document", simplifying and flattening the schema
+- Ease of migration for Desktop from JavaScript objects to YAML objects
+
+Cons:
+
+- Yet another data representation language to deal with
+- Readability over other formats is marginal
+
+## Examples
+
+To illustrate the YAML format, here is an example of the "homescreen" feature, first in JSON, then the same representation in YAML.
+
+### JSON Example
+
+This example is only intended to illustrate the structure of the data, and not serve as an example of the suggested format. See the YAML example below for the suggested format.
+
+```JSON
+{
+  "enums": [
+    {
+      "name": "HomeScreenSection",
+      "doc": "The sections of the homescreen",
+      "variants": [
+        {
+          "name": "top-sites",
+          "doc": "The original frecency sorted sites"
+        },
+        {
+          "name": "jump-back-in",
+          "doc": "Jump back in section"
+        },
+        {
+          "name": "recently-saved",
+          "doc": "Tabs that have been bookmarked recently"
+        },
+        {
+          "name": "recent-explorations",
+          "doc": "Tabs from another source"
+        },
+        {
+          "name": "pocket",
+          "doc": "Tabs from another source"
+        }
+      ]
+    }
+  ],
+  "objects": [],
+  "hints": {},
+  "features": [
+    {
+      "name": "homescreen",
+      "doc": "Represents the homescreen feature",
+      "props": [
+        {
+          "name": "sections-enabled",
+          "doc": "A map of booleans",
+          "type": {
+            "EnumMap": [
+              {
+                "Enum": "HomeScreenSection"
+              },
+              "Boolean"
+            ]
+          },
+          "default": {
+            "jump-back-in": false,
+            "pocket": false,
+            "recent-explorations": false,
+            "recently-saved": false,
+            "top-sites": true
+          }
+        },
+        {
+          "name": "section-ordering",
+          "doc": "The ordering of the sections on the homescreen",
+          "type": {
+            "List": {
+              "Enum": "HomeScreenSection"
+            }
+          },
+          "default": [
+            "top-sites",
+            "jump-back-in",
+            "recently-saved"
+          ]
+        }
+      ],
+      "default": null
+    }
+  ]
+}
+
+```
+
+### YAML Example
+
+This example is meant to illustrate the concise format and readability of YAML.
+
+```YAML
+---
+enums:
+- name: HomeScreenSection
+  doc: The sections of the homescreen
+  variants:
+  - name: top-sites
+    doc: The original frecency sorted sites
+  - name: jump-back-in
+    doc: Jump back in section
+  - name: recently-saved
+    doc: Tabs that have been bookmarked recently
+  - name: recent-explorations
+    doc: Tabs from another source
+  - name: pocket
+    doc: Tabs from another source
+objects: []
```

Comment:
what are objects/hints for?

Reply:
But, as said in other comments, this is a serialization of our IR, not what the user is going to type.

Reply:
Yes, I should have been more clear that this was just a draft yet and I haven't worked out the front-end schema completely.